JBBCode/Parser.php

```diff
@@ -246,16 +240,29 @@ public function codeExists($tagName, $usesOption = false)
      */
     public function getCode($tagName, $usesOption = false)
     {
-        foreach ($this->bbcodes as $code) {
-            if (strtolower($tagName) == $code->getTagName() && $code->usesOption() == $usesOption) {
-                return $code;
-            }
+        if($this->codeExists($tagName, $usesOption)) {
+            return $this->bbcodes[$this->getTagKey($tagName,$usesOption)];
         }
 
         return null;
     }
 
     /**
+     * Returns a key for the given combination of tag name and usesOption parameter
+     *
+     * @param string  $tagName    the name of the tag for which a key is to be generated
+     * @param boolean $usesOption whether or not the given tag accepts an option
+     * @return string a key representing tag and option
+     */
+    protected function getTagKey($tagName, $usesOption = false)
+    {
+        return serialize(array(
```

Comment:
What about using json_encode() instead of serialize()? It might be faster I think.